Update SDK versioning document to include support matrix guidance (#51412)
dotnet/docs: docs/core/porting/versioning-sdk-msbuild-vs.md

To ensure consistent tooling, you should use `dotnet build` rather than `msbuild` to build your application when possible.

## SDK and Visual Studio support matrix

While most developers use the .NET SDK bundled with their Visual Studio version, some configurations involve mismatched SDK and Visual Studio versions. We cannot ensure every Visual Studio version works with every in-support .NET SDK but rather test the most common configurations and make a best effort to ensure compatibility.

### Backward and forward compatibility

- **Primary supported configuration**: Use the SDK version bundled with your Visual Studio installation
    - Visual Studio 17.14 and .NET 9.0.3xx
    - Visual Studio 18.0 and .NET 10.0.1xx
- **Backward compatibility**: Using the latest feature band of the previous SDK version (for example, 8.0.4xx in Visual Studio 17.14) is supported with best-effort compatibility. We'll determine when to backport fixes based on risk and customer impact. 
While older SDKs than the latest of the prior band may work, we will not test them or make an effor to ensure any compatibility.

> - **Forward compatibility**: Using a newer SDK (for example, .NET 10 SDK in Visual Studio 17.14) is allowed without blocking, but targeting newer runtimes in older Visual Studio versions is not supported and should provide a build warning.

Reviewer: ```suggestion
- **Forward compatibility**: Using a newer SDK (for example, .NET 10 SDK in Visual Studio 17.14) is allowed without blocking, but targeting newer runtimes in older Visual Studio versions isn't supported and produces a build warning.
```

### Guidance

> Our guidance is to use the SDK that came with your Visual Studio instance (or latest available in CI or Visual Studio Code) as that will always have newest features including security updates.

Reviewer: ```suggestion
Use the SDK that came with your Visual Studio instance (or latest available in CI or Visual Studio Code), as that version always has the newest features, including security updates.
```

## Downlevel targeting support

The .NET SDK maintains targeting support for out-of-support .NET versions. The 10.0.100 SDK can build apps targeting `net9.0` all the way down to `netcoreapp1.0`.

### Guiding principles

> - **Existing support preserved**: Targeting support for out-of-support versions remains in the SDK without active removal. Customers upgrading their SDK or Visual Studio version should not have their builds broken simply by targeting an older .NET version.

Reviewer: ```suggestion
- **Existing support preserved**: Targeting support for out-of-support versions remains in the SDK without active removal. Customers that upgrade their SDK or Visual Studio version should not have their builds broken simply by targeting an older .NET version.
```
